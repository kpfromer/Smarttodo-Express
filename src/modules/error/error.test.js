import errorMiddleware from './error';
import boom from 'boom';
import express from 'express';
import request from 'supertest';

describe('error', () => {
  let initApp;
  let app;
  beforeEach(() => {
    initApp = error => {
      app = express();
      app.get('/error', (req, res, next) => 
        next(error)
      );
      app.use(errorMiddleware);
      return app;
    }
  });
  it('displays custom errors as json', async () => {
    const res = await request(initApp(boom.badRequest('Bad user input'))).get('/error');
    expect(res.status).toBe(400);
    expect(res.body).toMatchSnapshot();
  });
});