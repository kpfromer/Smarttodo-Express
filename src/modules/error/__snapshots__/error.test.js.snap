// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error displays custom errors as json 1`] = `
Object {
  "error": "Bad Request",
  "message": "Bad user input",
  "statusCode": 400,
}
`;
